server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -1135,6 +1145,13 @@ private String validateConfigurationValue(final String name, String value, final
             return errMsg;
         }
 
+        if (VirtualMachineManager.ResourceCountRoutersType.key().equalsIgnoreCase(name)
+                && (!resourceCountRoutersTypeValues.contains(value) || isBlank(value))) {
+            final String msg = "Possible values are: delta or all.";
+            s_logger.error(msg);
+            throw new InvalidParameterValueException(msg);
+        }
```

Comment:
please extract this to a method with a clear name, something that explains why the check is needed.